pipeline_products_session/jwst-data-products-part2.ipynb

```diff
@@ -0,0 +1,1401 @@
+    "    \n",
+    "   \n",
+    "* (optional) **2D or 3D background-subtracted data**\n",
+    "    * **Data model**: ImageModel or CubeModel\n",
+    "    * **File suffix**: ```_bsub``` or ```_bsubints```\n",
+    "    * **Description**: This is an intermediate product that is only created if the ```–save_bsub``` parameter is set to True. It will contain the data as output from the background step. If the input is a ```_rate``` product, this will be a ```_bsub``` product, while ```_rateints``` inputs will be saved as ```_bsubints```.  \n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's take a quick look at the standard imaging data product from Stage 2 that is used as input to Stage 3 processing: "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cal_image = datamodels.ImageModel(cal_file)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "create_image(cal_image.data[:,:], 0, 10, title=\"2D calibrated image\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cal_image.info()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Notice the arrays that are available in the calibrated images above:\n",
+    "\n",
+    "* ```area```\n",
+    "* ```data```\n",
+    "* ```dq```\n",
+    "* ```error```\n",
+    "* ```var_flat```\n",
+    "* ```var_poisson```\n",
+    "* ```var_rnoise```\n",
+    "\n",
+    "Also notice the ```bunit_data``` and ```bunit_err``` metadata values - those provide the units for the data. The data arrays can be accessed in the way we described before:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "variance_flat = cal_image.var_flat"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "variance_flat"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You'll notice in the metadata that there is more information -- for example, the association file name, data units, and WCS information. We'll revisit the WCS in the last section of this module. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "cal_image.meta.instance"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 6.2.<font color='white'>-</font>Spectroscopy<a class=\"anchor\" id=\"s2-spectroscopy\"></a>\n",
```

Comment:
You might want to define "configuration" here, or perhaps re-word to indicate that there are different sets of pre-defined steps that are mode dependent.

Reply:
Fixed